app/controllers/api/v1/coordinates_controller.rb

```diff
@@ -0,0 +1,14 @@
+class Api::V1::CoordinatesController < ApplicationController
+  def index
+    address = params[:address]
+
+    facade = CoordinatesFacade.new(address: address)
+    coordinates = facade.coordinates
+
+    if coordinates.present?
+      render json: { coordinates: coordinates }
+    else
+      render json: { error: 'Coordinates not found' }, status: :not_found
+    end
```

Comment:
I could be misreading, but since your manual serializer ALWAYS returns something (`{ error: "Coordinates not found" }` if the response is empty), won't the service always return the coordinates json even if it errored? So it would end up looking like:
```
{
  coordinates: {
    error: "Coordinates not found"
  }
}

Reply:
Nice catch! I updated the response to just simply include the lat and lon if it is successful and to simply include the error and its message if it is not successful.